File: fixtures/scale/objc/large/Families/XorStatic/XorStaticDescriptor.m
#import "XorStaticDescriptor.h"
#import "XorStaticCipher.h"
#import "XorStaticVectors.h"

@implementation XorStaticDescriptor

- (NSString *)family {
    return @"xorstatic";
}

- (NSString *)suite {
    return @"stream";
}

- (id<CLCipher>)cipher {
    return [[XorStaticCipher alloc] init];
}

- (NSArray *)vectors {
    return [XorStaticVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/XorStatic/XorStaticDescriptor.php
<?php

namespace CipherLab\Families\XorStatic;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together XorStaticCipher and
// XorStaticVectors under the "stream" suite.
class XorStaticDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'xorstatic';
    }

    public function suite(): string
    {
        return 'stream';
    }

    public function cipher(): Cipher
    {
        return new XorStaticCipher();
    }

    public function vectors(): array
    {
        return XorStaticVectors::all();
    }
}
